### DISCUSSION TITLE
import tkinter error

### DISCUSSION THREAD

**Discussion by SXN31** [mitmproxy/mitmproxy#5059] [Q&A] [answered]
I had the idea to build a GUI in order to represent some of the data I'm getting from requests, as well as to enable/disable options which will impact how requests are modified.

However, I noticed that when I attempted to import tkinter I would consistently get an error: No module named 'tkinter'.

If I run a separate python script with just the tkinter initialization it runs fine, so it's not that tkinter is missing.

I also tried adding it to the load event hook which at least gave a bigger traceback but the same error.

![image](https://user-images.githubusercontent.com/43016545/149592517-336cec1a-58ae-457c-86d1-87e4b6ae4e61.png)

I've tried this two ways. first, just as a standard import at the top of the script, and second in the load event hook.

This is all being done from a Windows 10 VM with Python 3.

As a side note, I tried using PyQt5 and also got the import errors. Are there restrictions on what can be imported, or is there a better way of going about this?

**Comment by mhils** [comment, marked as answer]
How did you install mitmproxy? If you have mitmproxy installed from PyPI, importing tkinter should work just fine. The mitmproxy binaries on the other hand include their own Python environment (which would not pick up your local tkinter installation).

**Reply by SXN31** [reply]
I installed via the binaries so that might be it. I'll confirm and if so mark as answer, thanks!
